>From 595a9e84d6c09693d6faa164204acca21c10bfbe Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Mon, 2 Apr 2012 09:53:33 +0200
Subject: [PATCH] Initialize dbus-glib threading

Based on upstreams 990e10860a7a624edcc38241422b7ad39f2cdf45, fixes gnome
bug #659756.
---
 addressbook/libedata-book/Makefile.am           |    7 +++++--
 addressbook/libedata-book/e-data-book-factory.c |    5 +++++
 calendar/libedata-cal/Makefile.am               |    6 ++++--
 calendar/libedata-cal/e-data-cal-factory.c      |    5 +++++
 configure.ac                                    |    9 +++++++++
 5 files changed, 28 insertions(+), 4 deletions(-)

diff --git a/addressbook/libedata-book/Makefile.am b/addressbook/libedata-book/Makefile.am
index 2fff48a..6af90c3 100644
--- a/addressbook/libedata-book/Makefile.am
+++ b/addressbook/libedata-book/Makefile.am
@@ -90,7 +90,8 @@ e_addressbook_factory_CPPFLAGS = \
 	-I$(top_builddir)/addressbook \
 	$(EVOLUTION_ADDRESSBOOK_CFLAGS) \
 	$(GOA_CFLAGS) \
-	$(FACTORY_GTK_CFLAGS)
+	$(FACTORY_GTK_CFLAGS) \
+	$(DBUS_GLIB_CFLAGS)
 
 e_addressbook_factory_SOURCES = \
 	e-data-book-factory.c \
@@ -103,7 +104,9 @@ e_addressbook_factory_LDADD = \
 	$(top_builddir)/libebackend/libebackend-1.2.la \
 	$(EVOLUTION_ADDRESSBOOK_LIBS) \
 	$(GOA_LIBS) \
-	$(FACTORY_GTK_LIBS)
+	$(FACTORY_GTK_LIBS) \
+	$(DBUS_GLIB_LIBS)
+
 
 %-$(API_VERSION).pc: %.pc
 	 cp $< $@
diff --git a/addressbook/libedata-book/e-data-book-factory.c b/addressbook/libedata-book/e-data-book-factory.c
index dca4255..e7dec12 100644
--- a/addressbook/libedata-book/e-data-book-factory.c
+++ b/addressbook/libedata-book/e-data-book-factory.c
@@ -26,6 +26,8 @@
 #include <unistd.h>
 #include <glib/gi18n.h>
 
+#include <dbus/dbus-glib.h>
+
 #ifdef ENABLE_MAINTAINER_MODE
 #include <gtk/gtk.h>
 #endif
@@ -842,6 +844,9 @@ main (gint argc,
 	gtk_init_check (&argc, &argv);
 	#endif
 
+	/* this is to initialize threading for dbus-glib used by GConf */
+	dbus_g_thread_init ();
+
 	context = g_option_context_new (NULL);
 	g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
 	g_option_context_parse (context, &argc, &argv, &error);
diff --git a/calendar/libedata-cal/Makefile.am b/calendar/libedata-cal/Makefile.am
index a7d1a02..11bf702 100644
--- a/calendar/libedata-cal/Makefile.am
+++ b/calendar/libedata-cal/Makefile.am
@@ -107,7 +107,8 @@ e_calendar_factory_CPPFLAGS = \
 	-I$(top_builddir)/calendar			\
 	$(LIBICAL_CFLAGS)				\
 	$(EVOLUTION_CALENDAR_CFLAGS)			\
-	$(FACTORY_GTK_CFLAGS)
+	$(FACTORY_GTK_CFLAGS) \
+	$(DBUS_GLIB_CFLAGS)
 
 e_calendar_factory_LDADD =						\
 	$(top_builddir)/calendar/libecal/libecal-1.2.la			\
@@ -117,6 +118,7 @@ e_calendar_factory_LDADD =						\
 	$(top_builddir)/libebackend/libebackend-1.2.la			\
 	$(LIBICAL_LIBS)							\
 	$(EVOLUTION_CALENDAR_LIBS)					\
-	$(FACTORY_GTK_LIBS)
+	$(FACTORY_GTK_LIBS) \
+	$(DBUS_GLIB_LIBS)
 
 -include $(top_srcdir)/git.mk
diff --git a/calendar/libedata-cal/e-data-cal-factory.c b/calendar/libedata-cal/e-data-cal-factory.c
index df3e136..a831d64 100644
--- a/calendar/libedata-cal/e-data-cal-factory.c
+++ b/calendar/libedata-cal/e-data-cal-factory.c
@@ -30,6 +30,8 @@
 #include <unistd.h>
 #include <glib/gi18n.h>
 
+#include <dbus/dbus-glib.h>
+
 #ifdef ENABLE_MAINTAINER_MODE
 #include <gtk/gtk.h>
 #endif
@@ -1043,6 +1045,9 @@ main (gint argc,
 	gtk_init_check (&argc, &argv);
 	#endif
 
+	/* this is to initialize threading for dbus-glib used by GConf */
+	dbus_g_thread_init ();
+
 	context = g_option_context_new (NULL);
 	g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
 	g_option_context_parse (context, &argc, &argv, &error);
diff --git a/configure.ac b/configure.ac
index 3d3995e..0bdba2c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -345,6 +345,15 @@ if `$PKG_CONFIG --atleast-version=0.9.1 libgdata`; then
 	AC_DEFINE(HAVE_LIBGDATA_0_9, 1, [libgdata is 0.9.1 or higher])
 fi
 
+dnl ******************************
+dnl DBus-glib stuff, to initialize thread for GConf, which is using it
+dnl this is returned back only temporarily and wil lbe removed as soon
+dnl as eds will stop using GConf completely
+dnl ******************************
+PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1 >= 0.6)
+AC_SUBST(DBUS_GLIB_CFLAGS)
+AC_SUBST(DBUS_GLIB_LIBS)
+
 dnl *******************************
 dnl Check for GNOME Online Accounts
 dnl *******************************
-- 
1.7.9.5

